### PR TITLE
Encrypt images

### PR DESCRIPTION
**Pull request by philipandersson**
closes denotedxyz/issues#23

### CODE REVIEW

File: apps/web/src/components/PublishMenu.tsx [hunk 1]
```diff
@@ -37,6 +54,8 @@ export const PublishMenu: React.FC<PublishmenuProps> = ({ page }) => {
 
   const publishMutation = useMutation(
     async () => {
+      page.data = await onPublishContent(page.data, encryptionKey);
```
Comment (ericzakariasson) [inline apps/web/src/components/PublishMenu.tsx:57]: I suspect this mutation of props can introduce annoying bugs 🐞 maybe we can make it immutable?

**Review by ericzakariasson — APPROVED** [review]
Great work 👏

File: apps/web/src/components/PublishMenu.tsx [hunk 2]
```diff
@@ -10,6 +10,22 @@ import { Popover, PopoverContent, PopoverTrigger } from "./ui/popover";
 import { Button, buttonVariants } from "./ui/button";
 import { cn } from "../lib/utils";
 import { useToast } from "./ui/use-toast";
+import * as IpfsImage from './commands/ipfs-image';
+import { JSONContent } from "@tiptap/react";
+
+async function onPublishContent(blocks: JSONContent[], encryptionKey: CryptoKey | undefined) : Promise<JSONContent[]> {
```
Comment (Kenny-Gin1) [inline apps/web/src/components/PublishMenu.tsx:16]: In a long and nested page, could the recursive functionality cause any problems? If there is, then I guess some error handling. If not, then not.

File: apps/web/src/components/PublishMenu.tsx [hunk 3]
```diff
@@ -37,9 +54,13 @@ export const PublishMenu: React.FC<PublishmenuProps> = ({ page }) => {
 
   const publishMutation = useMutation(
     async () => {
+      const pageToPublish = structuredClone(page);
```
Comment (Kenny-Gin1) [inline apps/web/src/components/PublishMenu.tsx:57]: Very cool!

**Review by Kenny-Gin1 — COMMENTED** [review]
Very nice!

File: apps/web/src/components/PublishMenu.tsx (hunk 2 above)
Comment (philipandersson) [reply apps/web/src/components/PublishMenu.tsx:16]: Yes we could exceed the maximum call stack size if the recursion is too deep but there is no way to recover from it. So I think is better that we just let it fail and have it reported to Sentry.

We would need to apply a more sophisticated method if we exceed call stack. Right now I'm awaiting each level to mitigate to many parallel calls at least.

**Pull request merged**